### PR TITLE
feat: Image view dialog implementation in product view page

### PR DESCRIPTION
**Pull request by gcharis**

### CODE REVIEW

File: src/app/features/products/product-view/product-view.component.html [hunk 1]
```diff
@@ -26,7 +26,9 @@
           </div>
 
           <div class="view__slideshow-indicator" fxLayoutAlign="center center">
-            View {{ slideshow?.slideIndex + 1 }} of {{ productForm.get('images').value.length }}
+            <button type="button" mat-button (click)="showImage(slideshow)">
+              View {{ slideshow?.slideIndex + 1 }} of {{ productForm.get('images').value.length }}
```
Comment (momegas) [inline src/app/features/products/product-view/product-view.component.html]: Why use string notation? I think you can do `productForm.images` or something similar.

File: src/app/features/products/product-view/product-view.component.ts [hunk 2]
```diff
@@ -8,18 +8,21 @@ import {
...
-  UploadProductImage,
+  UploadProductImage
 } from '../state/products.actions';
 import { ProductsState } from '../state/products.state';
 import { DbE, Product } from '@models';
 import { numberValidator } from '@validators';
 import { MediaState } from '../../media/state/media.state';
 import { BreakpointObserver, Breakpoints } from '@angular/cdk/layout';
+import { MatDialog } from '@angular/material';
+import { ViewImageComponent } from 'src/app/shared/view-image/view-image.component';
```
Comment (momegas) [inline src/app/features/products/product-view/product-view.component.ts]: use relative path

Comment (gcharis) [reply src/app/features/products/product-view/product-view.component.ts]: why

File: src/app/features/products/product-view/product-view.component.html (hunk 1 above)
Comment (gcharis) [reply src/app/features/products/product-view/product-view.component.html]: Can you be more specific?

The slideshow api gives the image that is being viewed

File: src/app/features/products/product-view/product-view.component.ts (hunk 2 above)
Comment (momegas) [reply src/app/features/products/product-view/product-view.component.ts]: Because you use it in the other imports as well :P

**Pull request merged**